content/blog/2020/07/2020-07-03-github-checks-api-plugin-coding-phase-1.adoc

```diff
@@ -0,0 +1,106 @@
+
+== Features from Coding Phase 1
+
+In the past month, our team was mostly working on the general checks API and an implementation for GitHub checks API.
+
+=== General Checks API
+
+Although the general checks API is developed based on the semantic meaning of GitHub Checks API, we still want to prepare it for similar concepts on other platforms like link:https://docs.gitlab.com/ee/api/commits.html#commit-status[Commit Status API] from GitLab.
+Contributions for implementations on these platforms are always welcomed!
+
+=== GitHub Checks API Implmentation
+
+Since we now have implemented the support for GitHub Checks API, plugin developers can easily create a GitHub check run for a GitHub branch source project by consuming our API.
+
+**Example:** To create a check run like:
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/created-check-run.png[Created Check Run]
+
+Consumers just need to use our API in this way:
+
+[source, java]
+----
+
+ChecksDetails details = new ChecksDetailsBuilder()
+        .withName("Jenkins")
+        .withStatus(ChecksStatus.COMPLETED)
+        .withDetailsURL("https://ci.jenkins.io")
+        .withStartedAt(LocalDateTime.now(ZoneOffset.UTC))
+        .withCompletedAt(LocalDateTime.now(ZoneOffset.UTC))
+        .withConclusion(ChecksConclusion.SUCCESS)
+        .withOutput(new ChecksOutputBuilder()
+                .withTitle("Jenkins Check")
+                .withSummary("# A Successful Build")
+                .withText("## 0 Failures")
+                .withAnnotations(Arrays.asList(
+                        new ChecksAnnotationBuilder()
+                                .withPath("Jenkinsfile")
+                                .withLine(1)
+                                .withAnnotationLevel(ChecksAnnotationLevel.NOTICE)
+                                .withMessage("say hello to Jenkins")
+                                .withStartColumn(0)
+                                .withEndColumn(20)
+                                .withTitle("Hello Jenkins")
+                                .withRawDetails("a simple echo command")
+                                .build(),
+                        new ChecksAnnotationBuilder()
+                                .withPath("Jenkinsfile")
+                                .withLine(2)
+                                .withAnnotationLevel(ChecksAnnotationLevel.WARNING)
+                                .withMessage("say hello to GitHub Checks API")
+                                .withStartColumn(0)
+                                .withEndColumn(30)
+                                .withTitle("Hello GitHub Checks API")
+                                .withRawDetails("a simple echo command")
+                                .build()))
+                .build())
+        .withActions(Collections.singletonList(
+                new ChecksAction("re-run", "re-run Jenkins build", "#0")))
```

Comment:
isn't re-run built into the API? is it a realistic action?

Reply:
Just an example here, the action is not supported yet. I'll change it to something else to avoid misconceptions.